### PR TITLE
Disable StrictMode by default

### LINKED ISSUE #8586: Disable `StrictMode` by default and log a message when it is enabled
We use `StrictMode` to help us debug potential issues in development. It has been added in https://github.com/mozilla/addons-frontend/issues/6189.

In https://github.com/mozilla/addons-frontend/issues/6694, we introduce a config flag to enable/disable it but it is turned on by default in our config. `StrictMode` won't be executed in production because it respects `NODE_ENV=production`.

**Proposal 1:** I suggest we disable it by default and only enable it for local dev.

`StrictMode` triggers redux actions twice and it is confusing (sometimes). This behavior has been described in https://github.com/mozilla/addons-frontend/issues/6424.

**Proposal 2:** we should log a message when `StrictMode` is enabled and link to the issue https://github.com/mozilla/addons-frontend/issues/6424 so that we remember why we see redux actions dispatched twice.

### PR DESCRIPTION
**Pull request by alexkayabula**
Fixes #8586 

   - Disables `StrictMode` by default.
   - Enables `StrictMode` in local development.

### CODE REVIEW

**Comment by alexkayabula** [comment]
> Great start! I think we still need to log a message when `enableStrictMode` is `true`, though.

Yeah true, I have added that.

File: src/core/server/base.js [hunk 1]
```diff
@@ -579,6 +579,17 @@ export function runServer({
...
+
+            const strictModeEnabled = convertBoolean(
+              config.get('enableStrictMode'),
+            );
+
+            if (strictModeEnabled) {
+              log.info(
+                `🔥  StrictMode enabled, checkout this link https://github.com/mozilla/addons-frontend/issues/6424.`,
+              );
+            }
```
Comment (willdurand) [inline src/core/server/base.js]: Could you put that in `src/core/client/base.js` instead please? I think we do not need the flame (although I like it).

Could you put this message:

```
StrictMode is enabled, which  causes double redux action dispatching. See: https://github.com/mozilla/addons-frontend/issues/6424
```

File: src/core/client/base.js [hunk 2]
```diff
@@ -30,6 +30,14 @@ export default async function createClient(
     await fetchBufferedLogs();
   }
 
+  const strictModeEnabled = convertBoolean(config.get('enableStrictMode'));
```
Comment (willdurand) [inline src/core/client/base.js]: I don't think we need a call to `convertBoolean`, you can probably remove it and inline `config.get('enableStrictMode')` in the `if` statement.

File: src/core/client/base.js [hunk 3]
```diff
@@ -30,6 +30,14 @@ export default async function createClient(
...
+
+  if (strictModeEnabled) {
+    log.info(
+      `StrictMode is enabled, which  causes double redux action dispatching. See: https://github.com/mozilla/addons-frontend/issues/6424.`,
```
Comment (willdurand) [inline src/core/client/base.js]:
```suggestion
      `StrictMode is enabled, which causes double redux action dispatching. See: https://github.com/mozilla/addons-frontend/issues/6424`,
```

Comment (alexkayabula) [reply src/core/client/base.js]: Changes made

File: src/core/client/base.js (hunk 2 above)
Comment (alexkayabula) [reply src/core/client/base.js]: I have refactored and made the changes.

**Comment by willdurand** [comment]
awesome, thanks @alexkayabula

**Pull request merged**